addons/sale_quotation_builder/models/sale_order_template.py

```diff
@@ -20,5 +20,5 @@ def action_open_template(self):
         return {
             'type': 'ir.actions.act_url',
             'target': 'self',
-            'url': '/sale_quotation_builder/template/%d' % self.id
+            'url': '/@/sale_quotation_builder/template/%d' % self.id
```

Comment:
Seeing this, I checked around the code. Sometimes we use `get_client_action_url` sometimes we simply prefix by `/@`, and each time I am wondering if there is any difference. After testing, I think there is at least one big difference and this one here should probably use `get_client_action_url` instead: it would be very much faster not to go through the `/@` controller in this case as the page would not even reload: you would just trigger an action and staying in the same backend.

Note: I guess there are more example but while trying to find out where this action was triggered, I found another one: when going on a quotation (not a quotation template), you have a button "Customer Preview" on top of the form (button box), that's kinda the same as this one. Maybe this could be fixed in the same commit as well? And there too, I would use `get_client_action_url` and I would probably make sure the documentation of the method explains the difference with `/@` if you agree :+1:

Reply:
> it would be very much faster not to go through the /@ controller in this case as the page would not even reload: you would just trigger an action and staying in the same backend.

Well, I'd rather keep the `/@/` notation and make the framework respond to it as we want: without reloading anything and calling the website preview client action itself. It will be easier to read, to write and to remember IMHO. That should be feasible, right?

> Maybe this could be fixed in the same commit as well?

Sure, I actually created another commit as I found yet another occurrence we missed in invoicing. Those 2 (the one you found and this one) are a bit less critical as not fundamentally wrong (It is stating you will land on a preview, while the other states you will land on the builder to edit it), but should still be restored as it was in 15.3 and before.
Note that I don't really like the wording of those behaviors, especially the part where in the "preview" it says "Go back to edit mode", I feel like edit mode doesn't really mean anything, we should maybe align this to be in par with the rest of our wording, something like "Go back to settings" (or even better if we can retrieve the model name: "Go back to Invoice settings").
Less important but is it supposed to be easy to figure that you can actually click on "Preview" / "Customer Preview" to be able to edit the SO/Invoice and not only see the preview?
Maybe those could be renamed "Preview & Edit" or something.
cc @pierre-pde

Reply:
> That should be feasible, right?

It would not be feasible in all cases. So I would not implement it for one case and have others that don't work. You could somehow patch the backend method in charge of responding to python action in the case of website (not sure the framework team would be happy about that) and say "if it starts with `/@` and I am in the frontend, then it's not an act_url but going to the client action with ..."... that would work. But if someone does `window.location.href = '/@/machin'`, that will not work (and in that case, there is no way to make it work).

Note that having an `act_url` action which use `get_client_action_url` is not perfect either... maybe we could directly return the definitin of the client action rather than reaching its URL and let the backend parse the URL to then trigger the client action. We could have a method method `get_client_action(<url_param_here>)` returning the whole action description instead of returning its URL.

> Wording [...]

Whatever is fine to me.... I am not for "Preview & Edit" though. "Preview" is fine to me: you reach the client action "Preview" and you can edit (but maybe won't). Or then we should rename our client action "Website Preview & Edit" instead of "Website Preview" to be consistent.

Reply:
I was more speaking about the "Go back to Edit mode" to go back on a form view, which is confusing with our builder, more than the "Preview" as I mentioned, the "Preview"/"Customer Preview" are fine-ish.
But @pierre-pde created a task or something somewhere to take care of it at some point. Didn't yet find something very clear to replace it, but "Back to Edit Mode" to go back to a form view is really.. weird within a website builder

About the other part of your message, I'll have a look when I go back to this PR.

Reply:
> but "Back to Edit Mode" to go back to a form view is really.. weird within a website builder

Yep agreed, like I said any whatever is fine to me, just wanted to precise for the part "Preview & Edit" I was not a fan of.

Reply:
I've created task #2984959 to harmonize & improve the flow between form view & website view